appserver/connectors/connectors-runtime/src/main/java/com/sun/enterprise/connectors/deployment/annotation/handlers/ConnectionFactoryDefinitionHandler.java

```diff
@@ -268,7 +160,7 @@ private ConnectionFactoryDefinitionDescriptor createDescriptor(ConnectionFactory
         desc.setMaxPoolSize(defn.maxPoolSize());
         desc.setMinPoolSize(defn.minPoolSize());
 
-        if (defn.description() != null && !defn.description().equals("")) {
+        if (defn.description() != null && !defn.description().isEmpty()) {
```

Comment:
Could also use the GF utility method for this; 

com.sun.enterprise.util.Utility.isEmpty(String)

Reply:
> This is a large broad set of changes touching a number of subsystems and 212 files. Not sure I can review it tbh. I'd have to assume it is correct.

I completely agree that it is horrible, but I have no idea how to move from "swamps of DOL" by a different way. I could make a PR after every safe point, but even then it would not help too much. Everything is fragile, in previous iterations I learned some "rules" how to avoid breaking things. Also in the first iteration took whole day to find out where I did the bad step, this time it was much faster.

There are overlapping interfaces and methods, removing one can break deployment or related things despite it compiles. Many reflections used to set attributes or just call methods; when they are just setters or "adders", it is easy, but sometimes they depend on the order of execution, which is fragile, fragile, fragile ...

That leads to some "stack" and "routine" in refactoring. And also thinking about next iterations.

Reviewing it by reading code is nearly worthless (except well visible mistakes I really could make), that's why I always run the TCK and why we were helping the TCK (so we can run it against GF PR), in last two years we also (partially) fixed the "Zombie Army" (Ant tests, I'm often checking if the count of passing tests did not decrease), now I'm trying to make friends of GF and CargoTracker (the last issue: derby pool). 

Good thing is that in the future it should converge to some more ideal state and those PRs should be smaller and smaller. I believe that creating them is still more painful than reading them.